File: 2022/gi.rb
require 'httparty'
require 'fileutils'

cookie = ENV['AOC_SESSION']
base_url =  'adventofcode.com'
options = { headers: { 'Cookie' => "session=#{cookie}" } }
day = ARGV[0]
year = ARGV[1] || 2022
response = HTTParty.get("http://adventofcode.com/#{year}/day/#{day}/input", options)
directory = File.expand_path("../#{year}/inputs")
unless File.directory?(directory)
    FileUtils.mkdir(directory)
end
File.open("#{directory}/#{day}.txt", 'w') { |f| f.write(response.body) }

File: 2020/gi.py
import requests
import sys
import os

if len(sys.argv) < 2:
    raise Exception("Pass in a day number so we know what day to download.")

day_number = sys.argv[1]

input_contents = requests.get(
    f"https://adventofcode.com/2020/day/{day_number}/input",
    cookies={"session": os.environ.get("AOC_SESSION")},
).text

with open(f"inputs/{day_number}.txt", "w") as file:
    file.write(input_contents)
